### PR TITLE
Add evaluate method to the background 3D

### PR DESCRIPTION
**Pull request by JouvinLea**
- The evaluate method is based on the new evaluate method in NDdata array from a set of coordinates.
- Clean a bit the test of background 3D with a more simple test

### CODE REVIEW

File: gammapy/irf/background.py [hunk 1]
```diff
@@ -134,6 +134,31 @@ def to_fits(self, name='BACKGROUND'):
...
+        """
+        Evaluate the `Background3D` at a given FOV coordinate and energy. The coordinates det_x, det_y and erngy_reco
+        should have the same dimension that match the numbers of point you want to evaluate
+        Parameters
+        ----------
+        detx : `~astropy.coordinates.Angle`
+                FOV coordinate X-axis binning. Same dimension than det_y and energy_reco
+        dety: `~astropy.coordinates.Angle`
+                FOV coordinate Y-axis binning. Same dimension than det_x and energy_reco
+        energy_reco : `~astropy.units.Quantity`
```
Comment (registerrier) [inline gammapy/irf/background.py]: You might use names fov_lon and fov_lat to comply with  https://github.com/gammapy/gammapy/blob/f91338a966a6c935f6ae66f36de2c838254b487a/docs/development/howto.rst

File: gammapy/irf/tests/test_background.py [hunk 2]
```diff
@@ -21,35 +31,65 @@ def test_background_3d_basics(bkg_3d):
...
+    assert data.unit == 's-1 MeV-1 sr-1'
+
+
 @requires_dependency('scipy')
 @requires_data('gammapy-extra')
 def test_background_3d_evaluate(bkg_3d):
-    bkg_rate = bkg_3d.data.evaluate(energy='1 TeV', detx='0.2 deg', dety='0.5 deg')
-    assert_allclose(bkg_rate.value, 0.00013352689711418575)
-    assert bkg_rate.unit == 's-1 MeV-1 sr-1'
+    # Evaluate at nodes in energy
```
Comment (registerrier) [inline gammapy/irf/tests/test_background.py]: Shouldn't there still be a test of `bkg3d.data.evaluate`?

File: gammapy/irf/background.py (hunk 1 above)
Comment (cdeil) [reply gammapy/irf/background.py]: I'll do this before merging.

File: gammapy/irf/tests/test_background.py (hunk 2 above)
Comment (cdeil) [reply gammapy/irf/tests/test_background.py]: I would say no, a test for `bkg_3d.data` is not needed.
`.data` is an implementation detail that shouldn't be called.
The one existing caller should be adapted later today in a follow-up PR.

**Comment by registerrier** [comment]
OK now. Merging.

File: gammapy/irf/background.py [hunk 3]
```diff
@@ -134,6 +134,32 @@ def to_fits(self, name='BACKGROUND'):
         """Convert to `~astropy.io.fits.BinTable`."""
         return fits.BinTableHDU(self.to_table(), name=name)
 
+    def evaluate(self, detx, dety, energy_reco, method=None, **kwargs):
+        """
+        Evaluate the `Background3D` at a given FOV coordinate and energy. The coordinates det_x, det_y and erngy_reco
+        should have the same dimension that match the numbers of point you want to evaluate
```
Comment (cdeil) [inline gammapy/irf/background.py]: I think you also need an empty line here above the `Parameters` section for the docs build to work.

File: gammapy/irf/tests/test_background.py [hunk 4]
```diff
@@ -21,35 +31,65 @@ def test_background_3d_basics(bkg_3d):
...
+    res = bkg_3d.evaluate(detx=np.array([1, 0.5]) * u.deg, dety=np.array([1, 0.5]) * u.deg,
```
Comment (cdeil) [inline gammapy/irf/tests/test_background.py]: You can remove the `np.array` calls here and the result should be exactly the same, just a bit less code.

**Pull request merged**